client/src/components/admin/ResultForm/ResultForm.js

```diff
@@ -121,9 +121,29 @@ const ResultForm = ({
             initialValue={attempt}
             disabled={!result || index >= disabledFromIndex}
             onValue={value => {
-              const updatedValue =
-                timeLimit && value >= timeLimit.centiseconds ? -1 : value;
-              const updatedAttempts = setAt(attempts, index, updatedValue);
+              var updatedAttempts = setAt(attempts, index, value);
```

Comment:
Please use `const` everywhere to stay consistent across the project (either way `var` should be avoided since `let` and `const` introduction, as these are block-scoped and thus most predictable).

While on that, functional approach is preferable here (partially enforced by React, but I'd rather stick to it in general), so please avoid any sort of mutations and imperative-style computation ^^

On a side-note, as this is quite some logic now, it would be best to move it to another module (`logic/attempts.js`), here's how I'd go about it (iterating over the attempts, keeping track of the sum, and producing an updated list of attempts):
```js
const applyTimeLimit = (attempts, timeLimit) => {
  if (timeLimit.cumulativeRoundIds.length === 0) {
    return attempts.map(attempt =>
      attempt >= timeLimit.centiseconds ? -1 : attempt
    );
  } else {
    /* Note: for now cross-round cumulative time limits are handled
       as single-round cumulative time limits for each of the rounds. */
    const [updatedAttempts, sum] = attempts.reduce(
      ([updatedAttempts, sum], attempt) => {
        const updatedSum = attempt > 0 ? sum + attempt : sum;
        const updatedAttempt =
          attempt > 0 && updatedSum >= timeLimit.centiseconds ? -1 : attempt;
        return [updatedAttempts.concat(updatedAttempt), updatedSum];
      },
      [[], 0]
    );
    return updatedAttempts;
  }
};
```
(for cross-cumulative time limits it's still better to check them within each round, than do nothing)

Then here it would become
```js
const updatedAttempts = applyTimeLimit(
  setAt(attempts, index, value),
  timeLimit
);
```
As you can see all the functions return new arrays, keeping the arguments untouched.

If you feel like it, you can also define a similar `applyCutoff` function to move the cutoff logic outside =)

Also please write some tests for that in `logic/tests/attempts.test.js` (`meetsCutoff` may be a good example to base on).

Reply:
Hi Jonathan, thanks a lot for the detailed comments. It was really helpful.
I'm a bit confused because I couldn't find the code you wrote in the comment in the latest code. May I know whether you wrote it for me so that I can use that function?
If so, I'll write it and write a similar apply cutoff too as you said. Can I proceed?

Reply:
Hey, none of that code already exists! All I meant was to define the `applyTimeLimit` function in the `attemtps.js` module and then use it [here](https://github.com/thewca/wca-live/blob/d7666dd3fb69426587b207a8ba8aec474e9df077/client/src/components/admin/ResultForm/ResultForm.js#L124-L126). I suggested defining similar `applyCutoff` to move [this](https://github.com/thewca/wca-live/blob/d7666dd3fb69426587b207a8ba8aec474e9df077/client/src/components/admin/ResultForm/ResultForm.js#L128-L132) logic into the `attempts.js` module as well. [These](https://github.com/thewca/wca-live/blob/d7666dd3fb69426587b207a8ba8aec474e9df077/client/src/logic/tests/attempts.test.js#L143-L166) are the tests that may be a good start for testing the new functions =)